IA1.2.11 Write scripts and documentation for demo

shenkeyao opened the pull request:
Closes https://app.asana.com/1/1208976916964769/project/1209392461754458/task/1211039956508130?focus=true.

### This PR:
* Adds scripts for the demo with Celo to pre-build everything, start services, view logs, and shut down everything.
* Updates documentation about how to run scripts.

### Key places to review:
* Try the scripts following `README_ESPRESSO.md`.

<!-- ### How to test this PR:  -->
<!-- Optional, uncomment the above line if this is relevant to your PR -->
<!-- If your PR is fully tested through CI there is no need to add this section -->
<!-- * E.g. `just test` -->

<!-- ### Things tested -->
<!-- Anything that was manually tested (that is not tested in CI). -->
<!-- E.g. building/running of docker containers. Changes to docker demo, ... -->
<!-- Especially mention anything untested, with reasoning and link an issue to resolve this. -->

<!-- Complete the following items before creating this PR -->
<!-- [ ] Issue linked or PR description mentions why this change is necessary. -->
<!-- [ ] PR description is clear enough for reviewers. -->
<!-- [ ] Documentation for changes (additions) has been updated (added).  -->
<!-- [ ] If this is a draft it is marked as "draft".  -->
<!-- [ ] For integration projects, make sure it won't break backward compatibility. -->

<!-- Details on maintaining backward compatibility for integration projects: -->
<!-- * Try to keep changes additive: add new, optional methods, flags, or parameters instead of modifying or removing existing functionality. -->
<!-- * If modification is necessary, it should be either a clear bug fix or guarded by a config/feature flag.  -->
<!-- * Follow Open Closed Principle and Interface Segregation Principle, clients should not be forced to depend on interfaces they do not use. -->

<!-- To make changes to this template edit https://github.com/EspressoSystems/.github/blob/main/PULL_REQUEST_TEMPLATE.md -->

jbearer committed:
Add devnet test for batcher restart

jbearer committed:
Check error returns

jbearer committed:
Separate op-geth instances for each L2 node

jbearer committed:
Build devnet dockers in CI

jbearer committed:
Build op-deployer in CI

jbearer committed:
Try larger runner

jbearer committed:
Increase test outage and recovery time

jbearer committed:
Try to speed up transaction verification

jbearer committed:
Do not drop batches before we have seen a finalized L1 block

jbearer committed:
Remove unnecessary sleep

shenkeyao committed:
Add scripts for demo and documentation

jbearer committed:
Build containers in dependency order

jbearer committed:
Don't copy config file into Docker image at build time

shenkeyao committed:
Fix syntax

jbearer committed:
Checkout submodules in CI

Ayiga commented on espresso/scripts/logs.sh:
```diff
@@ -0,0 +1,31 @@
+    "caff-node"
+    "op-batcher"
+    "op-proposer"
+)
```
These don't seem to be utilized.  It seems like these are meant to validate the user input against, but no validation change is performed.

shenkeyao replied:
I originally planned to add input validation, but then decided to have a simple version working first. 😄 Can add it later!

shenkeyao replied:
Updated! Also added some service name aliases.

jbearer committed:
Don't copy config file into Docker image at build time

shenkeyao committed:
Remove another COPY

jbearer committed:
Run devnet test in separate workflow

Ayiga committed:
Add exposed ports for more nodes in docker-compose

The `verifier`, `caff-node`, and `sequencer` all do not have their ports forwarded to
the host machine.  Having these ports exposed can make testing / debugging easier
by utilizing the optimism RPC API.

This change exposes these ports.

Ayiga committed:
Fix CAFF environment variables

The CAFF environment variables utilized in the `docker-compose.yml` need to have
a prefix of `OP_NODE_`, otherwise they will not apply.  These not applying causes
the `caff-node` to never make progress.

This change corrects the misconfigured `caff-node` environment variables, and
cleans up an unnecessary one.

Ayiga committed:
Set `caff.node` to `true` for `caff-node`

The `caff-node` is meant to be running a derivation based on information retrieved
from Espresso's network.  However, it needs to be enabled to do so.  This change
enables the `caff-node`'s mode by explicitly setting `caff.node` to `true` in the
launch configuration.

shenkeyao committed:
Support alias and input check

shenkeyao committed:
Merge branch 'keyao/demo' of github.com:EspressoSystems/optimism-espresso-integration into keyao/demo

Ayiga committed:
Add scripts and instructions for running utility script

In order to showcase the speed and progress of the Espresso solution with the Caff
node, versus the existing sequencer, or the L1 derived verifier, a script has been
added that utilities `watch` and `tmux` to provide a nice visual comparison between
the three nodes.

shenkeyao committed:
Merge branch 'jb/ta1-batcher-restart' into keyao/demo

shenkeyao committed:
Update scripts after separating geth

shenkeyao committed:
Merge branch 'keyao/demo' of github.com:EspressoSystems/optimism-espresso-integration into keyao/demo

shenkeyao committed:
Merge branch 'celo-integration-rebase-13.2' into keyao/demo

shenkeyao committed:
Update readme

philippecamacho reviewed (approved):
LGTM

philippecamacho committed:
tmux in flake.nix

philippecamacho commented on README_ESPRESSO.md:
````diff
@@ -380,3 +380,87 @@ In order to refresh this AMI one needs to:
+After that has been done you should be able to spin up the simple script
+using the following command:
+```console
+./espresso/scripts/demo_tmux_get_sync_status.sh
````
When I run this script I get "sh: 1: ./get_sync_status.sh: not found ".

philippecamacho reviewed (changes requested):
The tmux script does not work for me.

Ayiga committed:
Update shell being targeted in get_sync_status.sh

The `get_sync_status.sh` script attempts to target `zsh` residing within `/bin/zsh`,
but this isn't a universally available. It is better to target a more commonly available
like `/bin/bash`.

This change modifies the script to target `/bin/bash` instead of `/bin/zsh`.

philippecamacho reviewed (approved):
LGTM
